airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/split_secrets/AWSSecretManagerPersistence.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.persistence.split_secrets;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.amazonaws.secretsmanager.caching.SecretCache;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.CreateSecretRequest;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretRequest;
+import com.amazonaws.services.secretsmanager.model.ResourceNotFoundException;
+import com.amazonaws.services.secretsmanager.model.UpdateSecretRequest;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * SecretPersistence implementation for AWS Secret Manager using <a href=
+ * "https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/secretsmanager/package-summary.html">Java
+ * SDK</a> The current implementation doesn't make use of `SecretCoordinate#getVersion` as this
+ * version is non-compatible with how AWS secret manager deals with versions. In AWS versions is an
+ * internal idiom that can is accessible, but it's a UUID + a tag <a href=
+ * "https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/secretsmanager/SecretsManagerClient.html#listSecretVersionIds--">more
+ * details.</a>
+ */
+@Slf4j
+public class AWSSecretManagerPersistence implements SecretPersistence {
+
+  private final AWSSecretsManager client;
+
+  @VisibleForTesting
+  protected final SecretCache cache;
+
+  /**
+   * Creates a AWSSecretManagerPersistence using the defaults client and region from the current AWS
+   * credentials. This implementation makes use of SecretCache as optimization to access secrets
+   *
+   * @see SecretCache
+   */
+  public AWSSecretManagerPersistence() {
+    this.client = AWSSecretsManagerClientBuilder.defaultClient();
```

Comment:
@mauricioalarcon before we get this PR merged, can you incorporate the AWS Secret Manager into https://github.com/airbytehq/airbyte/blob/92f71accea6674f04cabf1bcdc8a48f7f4866905/airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretPersistence.java similar to `GOOGLE_SECRET_MANAGER` and `VAULT`? 

We have two "modes" for secret persistence layers, "ephemeral" and "long-lived". For Google Secret Manager, we define separate TTLs for each, but for Vault, both use the same underlying implementation. I think for now, we can use the same implementation of the AWS Secret Manager for both, unless you want to take a look at the Google internals and replicate something similar for AWS, by all means go for it!

One final important note: these other secret persistence classes use credentials and configurations from our `Configs.java`, and I think it'd make sense to add some new configs for AWS Secrets Manager so that we're not relying on any defaults.

For example, these configs power the Vault-based secret persistence: https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-models/src/main/java/io/airbyte/config/Configs.java#L134-L150

And the implementation for these methods are here: https://github.com/airbytehq/airbyte/blob/master/airbyte-config/config-models/src/main/java/io/airbyte/config/EnvConfigs.java#L396-L409

This will make it easier for various environments to configure the AWS Secret manager, without needing to rely on default profiles, which can get particularly cumbersome in some environments where multiple profiles exist.

Let me know if you have any questions around any of that! Really appreciate your contribution here :)

Reply:
Thanks @pmossman all of these changes make sense. I'll work on these and send an update to the PR.

Reply:
@pmossman I adjusted the PR with the requested changes; I added the new config keys as part of the instructions at `docs/operator-guides/configuring-airbyte.md`, I'm not sure where else I need to update, if any.

As for the "ephemeral", that's not an option for AWS; that will require a bit more to mimic the TTL. I'll research a bit more on this, and I'll get back to you.

Thank you